### ISSUE TITLE
Custom eval_metric function ignored in calculation of validation score

### ISSUE THREAD

**Issue by davidmosca** [microsoft/LightGBM#2897]
## Environment info

Operating System: Windows 10

CPU/GPU model: Intel (R) Core(TM) i7-5600 CPU @ 2.60GHz

C++/Python/R version: Python 3.7.4

LightGBM version or commit hash: LightGBM version 2.3.0

## Error

In the code below, when metric==`'auc'` (second step in the loop), the only metric used for evaluation is `'auc'`, which is the expected behavour (output: Evaluated only: auc). However, when metric==`custom_metric` (third and last step in the loop), the only metric used for evaluation is `'binary_logloss'` and **not** `customer_metric` (output: Evaluated only: binary_logloss), which is **not** the expected behaviour.

## Reproducible examples

```
import numpy as np
from lightgbm import LGBMClassifier
from sklearn.metrics import roc_auc_score

def custom_metric(y_true, y_pred):
    return ('custom_metric', roc_auc_score(y_true, y_pred), True)

def test_metrics(metric):
    np.random.seed(22)
    verbose=True
    cl1 = LGBMClassifier()
    cl2 = LGBMClassifier(first_metric_only=True)
    SIZE = 10000
    X = np.random.uniform(0, 1, size=SIZE).reshape(-1, 1)
    Y = np.array([1 if (X[i] < 0.4 and np.random.binomial(1, 0.5)==1) else 0 for i in range(SIZE)])
    X_valid = np.random.uniform(0, 1, size=SIZE).reshape(-1, 1)
    Y_valid = np.array([1 if (X[i] < 0.4 and np.random.binomial(1, 0.5)==1) else 0 for i in range(SIZE)])
    valid_set = [(X_valid,Y_valid)]
    cl1.fit(X, Y, early_stopping_rounds=30, eval_metric='logloss', eval_set = valid_set, verbose=verbose, eval_names=['eval', 'train'])
    cl2.fit(X, Y, early_stopping_rounds=30, eval_metric=metric, eval_set = valid_set, verbose=verbose, eval_names=['eval', 'train'])
    Y1 = cl1.predict(X)
    Y2 = cl2.predict(X)
    roc1 = roc_auc_score(Y, Y1)
    roc2 = roc_auc_score(Y, Y2)
    print(roc1, roc2, roc1==roc2, metric)

for metric in ['logloss', 'auc', custom_metric]:
    test_metrics(metric)
```

## Steps to reproduce

1. Run code and read output.

**Comment by StrikerRUS** [comment]
@davidmosca You should disable default metric by setting `metric="None"`.
```
    cl2 = LGBMClassifier(metric="None", first_metric_only=True)
```
Refer to https://lightgbm.readthedocs.io/en/latest/Parameters.html#metric.
